src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -801,17 +803,38 @@ public override int GetHashCode()
 
         public bool Equals(Guid g) => EqualsCore(this, g);
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static bool EqualsCore(in Guid left, in Guid right)
         {
-            ref int rA = ref Unsafe.AsRef(in left._a);
-            ref int rB = ref Unsafe.AsRef(in right._a);
+            if (Sse2.IsSupported)
+            {
+                var g1 = Unsafe.As<Guid, Vector128<byte>>(ref Unsafe.AsRef(left));
+                var g2 = Unsafe.As<Guid, Vector128<byte>>(ref Unsafe.AsRef(right));
+
+                if (Sse41.IsSupported)
+                {
+                    var xor = Sse2.Xor(g1, g2);
+                    return Sse41.TestZ(xor, xor);
+                }
+
+                var result = Sse2.CompareEqual(g1, g2);
+                return Sse2.MoveMask(result) == 0b1111_1111_1111_1111;
+            }
+
+            return SoftwareFallback(left, right);
 
-            // Compare each element
+            static bool SoftwareFallback(in Guid left, in Guid right)
```

Comment:
Re `in`: just to double-check the discussion in the issue.

Reply:
Happy to take direction on this either way. I have found in my performance tests thus far that passing a `Guid` by reference outperforms passing by value - I suspect because the JIT has difficulty keeping the `Guid` parameters in registers because they have 11 fields.